modules/repository-datastax/src/main/java/com/intuit/wasabi/repository/cassandra/impl/AssignmentStats.java

```diff
@@ -0,0 +1,100 @@
+package com.intuit.wasabi.repository.cassandra.impl;
+
+import com.intuit.wasabi.assignmentobjects.Assignment;
+import com.intuit.wasabi.experimentobjects.Bucket;
+import com.intuit.wasabi.experimentobjects.Experiment;
+import com.intuit.wasabi.repository.cassandra.accessor.count.HourlyBucketCountAccessor;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.intuit.wasabi.repository.cassandra.impl.CassandraAssignmentsRepository.NULL_LABEL;
+
+
+public class AssignmentStats {
+
+    private static HourlyBucketCountAccessor hourlyBucketCountAccessor;
+    private static DateFormat dayFormatter = new SimpleDateFormat("yyyy-MM-dd");
+    private static DateFormat hourFormatter = new SimpleDateFormat("HH");
+    private static Map<Integer, Map<String, AtomicInteger>> hourlyCountMap;
+    private static final Object lock = new Object();
+    private static final int UUID_LENGTH = 36;
+
+    public AssignmentStats() {
+        hourlyCountMap = new ConcurrentHashMap<>();
+        for (int hour = 0; hour <= 23; hour++) {
+            hourlyCountMap.put(hour, new ConcurrentHashMap<>());
+        }
+    }
+
+    public void incrementCount(Experiment experiment, Assignment assignment) {
+        int assignmentHour = getHour(assignment.getCreated());
+        Map<String, AtomicInteger> hourMap = hourlyCountMap.get(assignmentHour);
+        Experiment.ID id = experiment.getID();
+        Optional<Bucket.Label> labelOptional = Optional.ofNullable(assignment.getBucketLabel());
+        Bucket.Label bucketLabel = labelOptional.orElseGet(() -> NULL_LABEL);
+        AtomicInteger oldCount = hourMap.get(ExpBucket.getKey(id, bucketLabel));
+        if (oldCount == null) {
+            synchronized (lock) {
```

Comment:
@iizrailevsky This is to ensure that multiple threads do not change the oldCount value at the same time, skewing the final value. oldCount would be initialized to 1 twice by the same thread instead of properly incrementing to 2 and then to 3. I'll add a comment to clarify.